typegen/evaluation/normalize_types.py

```diff
@@ -0,0 +1,131 @@
+def normalize_type(type_hint: str) -> str:
```

Comment:
Perhaps you want to perform the normalization process on the AST directly? sorting a string directly removes the concept of types in types e.g. `list[int | tuple[str, bytes]]`

Reply:
But the core principal of this is fine

Reply:
The reason for sorting is due to the merge when generating the metric data: None | str | int should be same as str | int | None. The types collected by the file type hints collector are all normalized so that we can compare unions correctly.  Since the type hints collected are str, the normalization has to work on str.